crepecake/src/main/java/com/google/cloud/tools/crepecake/builder/BuildConfiguration.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.builder;
+
+import com.google.cloud.tools.crepecake.registry.DockerCredentialRetriever;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+/** Immutable configuration options for the builder process. */
+public class BuildConfiguration {
+
+  /** Enumerates the fields in the configuration. */
+  @VisibleForTesting
+  enum Fields {
```

Comment:
This looks like a weird way to use Enums. Since the values of "Fields" aren't related in any way. I would normally expect an enum to represent a range of values for a single type? Looks like we're just using it as a way to represent contants?

Perhaps just make them members of BuildConfiguration directly?

baseImageServerUrl... etc ?

Reply:
This was mostly to be able to avoid a large set of `if` statements in Builder#build, since the `EnumMap` allows us to loop over the enum types. Perhaps a normal class would be better? I think later on I will add a required flag to the configurations as well.